src/components/user/services/userProfileSrv.js

```diff
@@ -1,76 +1,83 @@
 'use strict';
 
 angular.module('znk.infra.user').service('UserProfileService',
-    function (InfraConfigSrv, StorageSrv) {
+    function ($q, ENV, AuthService, UserStorageService) {
         'ngInject';
-        var profilePath = StorageSrv.variables.appUserSpacePath + '/profile';
+
+        var _this = this;
 
         this.getProfile = function () {
-            return InfraConfigSrv.getGlobalStorage().then(function(globalStorage) {
-                return globalStorage.get(profilePath).then(function (profile) {
-                    if (profile && (angular.isDefined(profile.email) || angular.isDefined(profile.nickname))) {
-                        return profile;
-                    }
-                    return InfraConfigSrv.getUserData().then(function(authData) {
-                        var emailFromAuth = authData.password ? authData.password.email : '';
-                        var nickNameFromAuth = authData.auth ? authData.auth.name : emailFromAuth;
+            var authData = AuthService.getAuth();
+            var profilePath = 'users/' + authData.uid + '/profile';
```

Comment:
why not use 
var profilePath = StorageSrv.variables.appUserSpacePath + '/profile';

it will remove the need to AuthService and